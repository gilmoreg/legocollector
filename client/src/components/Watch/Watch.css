.Watch {
  margin: 1rem;
}