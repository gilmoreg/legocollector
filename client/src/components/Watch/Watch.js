import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import WatchHeader from './WatchHeader/WatchHeader';
import WatchBody from './WatchBody/WatchBody';
import WatchFooter from './WatchFooter/WatchFooter';

import './Watch.css';

export class Watch extends Component {
  constructor(props) {
    super(props);
    this.state = {
      collapsed: true,
    };
  }

  render() {
    return (
      <div className="Watch">
        <WatchHeader watch={this.props.watch} />
        { this.state.collapsed ?
          <div />
          :
          <WatchBody
            watch={this.props.watch}
          />
        }
        <WatchFooter
          click={() => this.setState({
            collapsed: !this.state.collapsed,
          })}
          collapsed={this.state.collapsed}
        />
      </div>
    );
  }
}

Watch.propTypes = {
  // dispatch: PropTypes.func.isRequired,
  watch: PropTypes.shape({
    title: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
    stock_levels: PropTypes.array.isRequired,
  }).isRequired,
};

export default connect()(Watch);
