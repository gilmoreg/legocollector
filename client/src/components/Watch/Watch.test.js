import React from 'react';
import { shallow } from 'enzyme';
import { Watch } from './Watch';

it('renders without crashing', () => {
  const wrapper = shallow(
    <Watch dispatch={()=>{}}/>
  );
  expect(wrapper).toMatchSnapshot();
});
