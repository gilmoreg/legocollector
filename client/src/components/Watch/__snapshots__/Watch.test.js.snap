// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Watch
    dispatch={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <WatchHeader
          watch={undefined}
        />,
        <div />,
        <WatchFooter
          click={[Function]}
          collapsed={true}
        />,
      ],
      "className": "Watch",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "function",
        "props": Object {
          "watch": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "function",
        "props": Object {
          "click": [Function],
          "collapsed": true,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <WatchHeader
            watch={undefined}
          />,
          <div />,
          <WatchFooter
            click={[Function]}
            collapsed={true}
          />,
        ],
        "className": "Watch",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {
            "watch": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {
            "click": [Function],
            "collapsed": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
